Marlin/Marlin_main.cpp

```diff
@@ -1456,6 +1454,8 @@ static void set_axis_is_at_home(const AxisEnum axis) {
   {
     current_position[axis] = base_home_pos(axis);
   }
+  if (axis == Z_AXIS)
+    current_position[axis] -= zprobe_zoffset - Z_PROBE_OFFSET_FROM_EXTRUDER;
```

Comment:
This makes all printers behave like Cartesians that home with the probe

Reply:
How will this affect Z homing to a min endstop on a Cartesian which also has a Z probe?

Reply:
Homing will reflect zprobe_zoffset changes just like "homing with a probe" on Cartesians that home with end-stop as well as on delta's. As such all printers that have a probe (and zprobe_zoffset) will behave the same way.